### DISCUSSION TITLE
Score Chart in Reviews

### DISCUSSION THREAD

**Discussion by ecyrb** [online-go/online-go.com#2645] [Ideas]
When using a review, it would be convenient to see the original game's score chart, so that you can easily find the moves that result in big predicted score changes (without having to switch tabs to the original game).

(Maybe there is a way to do this I'm not aware of?)

And if I wanted to go about implementing this to test what it would look like, how should I best proceed? I see that in reviews, an ```AIDemoReview``` object is used instead of the ```AIReview``` (which contains the ```AIReviewChart``` object).

**Comment by dexonsmith** [comment]
IMO, adding an optional `AIReviewChart` to `AIDemoReview` (optional: only if it's a review, not if it's a demo) seems like how I'd start.

@anoek, looks like you're the only one that's touched AIDemoReview.tsx — any specific reason you left the chart out?

Some concrete motivation for including it: @ecyrb and I have been finding when we review our games together, we need two windows open:
1. The game window, so we can use the `AIReviewChart` to find the moves that deserve review
2. The review window, so we have a shared view for exploring moves

This works, but it's awkward not having it all in one window.

(Also, I've never heard of this "Discussion" thing on GitHub... does anyone pay attention to these? Would it be better to have this conversation on Slack?)

**Reply by anoek** [reply]
> (Also, I've never heard of this "Discussion" thing on GitHub... does anyone pay attention to these? Would it be better to have this conversation on Slack?) 

I have a slight bias towards slack, but here's fine too.

**Comment by anoek** [comment]
No reason that's still valid, I think it'd be great if we could get that chart in there. (Historically, we initially only supported reviewing an entire game at once and since reviews are more dynamic, we couldn't support it. Later though, we added support for analyzing variations in game, which is basically all that's needed for supporting reviews from the back end.)

Skimming over things, you might get pretty close by just dropping in `AIReview` when you have a source `game_id` in a review. There's probably some code that's assuming the board is a game, but it might be relatively easy to tweak it to work with reviews these days.

One thing that comes to mind as something to watch out for is reviewing of ongoing games. In those cases, you won't have reviews yet for the game (nor do we want to trigger them to start, I'm not sure what will happen if we do off the top of my head.). Once the game concludes though those will start appearing.
